crates/bevy_asset/src/asset_server.rs

```diff
@@ -896,19 +896,19 @@ mod test {
         // asset is loading
         assert_eq!(LoadState::Loading, get_load_state(&handle, &app.world));
 
-        app.update();
+        app.world.run_schedule(Main);
```

Comment:
I think this is worse: we're leaking internal details and making a very common operation more verbose.

Can we keep `App::update` around and just call this on the main schedule label?

Reply:
I did not do any test code refactoring, so it is verbose. :sweat_smile:

Generally, `&mut World` is passed in the exclusive system, so it is basically never used in that form. Probably in the form of runner functions and test code only.

And this PR will no longer be used in the engine code. So I think the maintenance will be forgotten and `App::update` will stop working somewhere.

Reply:
Following up on @alice-i-cecile 's question, can we keep `App::update()`?

It is used in tests and examples, and replacing it with the four-line stanza makes it more difficult to write tests that need to run schedules.  Making testing more difficult should be avoided.  I say this after adding easily a dozen `app.update()` calls in the stepping PR for an example demonstrating stepping.

> I think the maintenance will be forgotten and App::update will stop working somewhere.

The same argument applies if we replace the dozens of the existing `App::update()` calls with four other lines.  The difference is that we'll only need to fix `App::update()` when it breaks.

I think we do need for some function to exist on `mut App` that is the equivalent of "run everything".